spec/controllers/supplejack_api/users_controller_spec.rb

```diff
@@ -11,36 +11,71 @@ module SupplejackApi
   describe UsersController, type: :controller do
     routes { SupplejackApi::Engine.routes }
 
-    let(:user) { FactoryGirl.create(:user, authentication_token: 'abc123', role: 'admin') }
+    context 'admin user' do
+      let(:user) { FactoryGirl.create(:admin_user, authentication_token: 'abc123') }
 
-    describe 'GET show' do
-      it 'should assign @user' do
-        get :show, id: user.id, api_key: 'abc123', format: :json
-        expect(assigns(:user)).to eq(user)
+      describe 'GET show' do
+        it 'should assign @user when api key belongs to an admin' do
+          get :show, id: user.id, api_key: 'abc123', format: :json
+          expect(assigns(:user)).to eq(user)
+        end
       end
-    end
 
-    describe 'POST create' do
-      it 'should create a new user' do
-        allow(RecordSchema).to receive(:roles) { { admin: double(:admin, admin: true) } }
-        expect(User).to receive(:create).with({ 'name' => 'Ben' }).and_return(user)
-        post :create, api_key: 'abc123', user: { name: 'Ben' }, format: 'json'
+      describe 'POST create' do
+        it 'should create a new user' do
+          allow(RecordSchema).to receive(:roles) { { admin: double(:admin, admin: true) } }
+          expect(User).to receive(:create).with({ 'name' => 'Ben' }).and_return(user)
+          post :create, api_key: 'abc123', user: { name: 'Ben' }, format: 'json'
+        end
+      end
+
+      describe 'PUT update' do
+        it 'updates the user attributes' do
+          user_params = FactoryGirl.attributes_for(:user, name: 'Richard')
+          patch :update, id: user.api_key, api_key: user.authentication_token, user: user_params
+          user.reload
+          expect(user.name).to eq 'Richard'
+        end
       end
-    end
 
-    describe 'PUT update' do
-      it 'updates the user attributes' do
-        user_params = FactoryGirl.attributes_for(:user, name: 'Richard')
-        patch :update, id: user.api_key, api_key: user.authentication_token, user: user_params
-        user.reload
-        expect(user.name).to eq 'Richard'
+      describe 'DELETE destroy' do
+        it 'destroys the user' do
+          delete :destroy, id: user.id, api_key: user.authentication_token
```

Comment:
Are you sure at this point that a user exists?

Reply:
hmm good question I haven't changed this test tho, I will double check

Reply:
done !